### PR TITLE
[Split Build] Add periodic and trunk CI for cuda builds

### PR DESCRIPTION
**Pull request by PaliC**

### CODE REVIEW

File: .github/workflows/periodic.yml [hunk 1]
```diff
@@ -294,3 +294,85 @@ jobs:
       build-environment: linux-focal-rocm6.1-py3.8
       docker-image: ${{ needs.linux-focal-rocm6_1-py3_8-build.outputs.docker-image }}
       test-matrix: ${{ needs.linux-focal-rocm6_1-py3_8-build.outputs.test-matrix }}
+
+  linux-focal-cuda12_1-py3_10-gcc9-split-build:
```
Comment (atalman) [inline .github/workflows/periodic.yml:298]: Do we need so many ? Maybe just add cuda12_1 ?

**Review by atalman — APPROVED** [review]
lgtm, maybe we should keep all of them now. However after some testing have only i cuda version.

**Comment by PaliC** [comment]
@atalman I think if we're planning on doing releases, we should test the entire matrix. I think moving them to periodic is reasonable as well

**Pull request closed without merging**